sys/posix/sockets/posix_sockets.c

```diff
@@ -276,16 +281,42 @@ static int socket_close(int socket)
     return res;
 }
 
-static ssize_t socket_read(int socket, void *buf, size_t n)
+static inline int socket_fstat(vfs_file_t *filp, struct stat *buf)
+{
+    (void)filp;
+    memset(buf, 0, sizeof(struct stat));
+    buf->st_mode |= (S_IFSOCK | S_IRWXU | S_IRWXG | S_IRWXO);
+    buf->st_blksize = 512;
```

Comment:
I know 512 is pretty arbitrary, but the POSIX spec does not provide any more info on that. If someone has a better number, please provide.